## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#2bfc2a2a-18ce-455f-b111-6f2f35101a32) |
| mint-dist-xl.sh | :x: [more...](#4028566a-83f1-40db-857e-d0248fcd3507) |
| mint-gateway-nas.sh | :x: [more...](#29a29f27-0e88-48b0-b327-d38c6aa43633) |
<a name="29a29f27-0e88-48b0-b327-d38c6aa43633"></a>
### 8287-f0daf49/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30256
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 653f2be2da2f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 40 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 46 seconds
(8/14) Running minio-dotnet tests ... done in 25 seconds
(9/14) Running minio-go tests ... done in 37 seconds
(10/14) Running minio-java tests ... done in 14 seconds
(11/14) Running minio-js tests ... FAILED in 14 seconds
{
  "name": "minio-js",
  "function": "fGetObject(bucketName, objectName, filePath)",
  "args": "bucketName:minio-js-test-e85fb216-82ca-4019-8c13-42fd896f57b0, objectName:datafile-65-MB, filePath:/tmp/datafile-65-MB.download",
  "duration": 845,
  "status": "FAIL",
  "error": "Error: Size mismatch between downloaded file and the object at /mint/run/core/minio-js/node_modules/minio/dist/main/minio.js:965:12 at nextTask (node_modules/minio/node_modules/async/dist/async.js:4578:13) at next (node_modules/minio/node_modules/async/dist/async.js:4586:13) at /mint/run/core/minio-js/node_modules/minio/node_modules/async/dist/async.js:325:20 at FSReqWrap.oncomplete (fs.js:123:15)"
}

Executed 10 out of 14 tests successfully.
```
<a name="4028566a-83f1-40db-857e-d0248fcd3507"></a>
### 8287-f0daf49/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30118
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d1189321f358:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 2 minutes and 12 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 57 seconds
(8/14) Running minio-dotnet tests ... done in 54 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 22 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 42 seconds
(11/14) Running minio-js tests ... done in 52 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 1 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "args": {
    "presigned_get_object_url": "http://72.28.97.54:30118/minio-py-test-c98c8000-f95f-4f2b-bb78-cefe45ff0a22/7422d98f-d4c5-445c-9422-ad798f65cebb?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190922%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190922T184850Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=18ae00b2289750fe36cf3e1aae6ac377368ea9339f16954bf88b00cd49c19068",
    "response.reason": "Forbidden",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>7422d98f-d4c5-445c-9422-ad798f65cebb</Key><BucketName>minio-py-test-c98c8000-f95f-4f2b-bb78-cefe45ff0a22</BucketName><Resource>/minio-py-test-c98c8000-f95f-4f2b-bb78-cefe45ff0a22/7422d98f-d4c5-445c-9422-ad798f65cebb</Resource><RequestId>15C6D733962D7B3F</RequestId><HostId>373075b4-3d31-45cd-b84d-d0933faaa0c4</HostId></Error>",
    "bucket_name": "minio-py-test-c98c8000-f95f-4f2b-bb78-cefe45ff0a22",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-22T18-24-07Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C6D733962D7B3F\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Sun, 22 Sep 2019 18:48:56 GMT\"]}}",
    "object_name": "7422d98f-d4c5-445c-9422-ad798f65cebb",
    "response.status": 403
  },
  "duration": 126,
  "status": "FAIL"
}

Executed 11 out of 14 tests successfully.
```
<a name="2bfc2a2a-18ce-455f-b111-6f2f35101a32"></a>
### 8287-f0daf49/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30839
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp aea14027c9e2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 5 seconds
(3/14) Running aws-sdk-php tests ... done in 54 seconds
(4/14) Running aws-sdk-ruby tests ... done in 14 seconds
(5/14) Running awscli tests ... done in 2 minutes and 13 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 22 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 50 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 19 seconds
{
  "args": {},
  "duration": 100593,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```

--- a/mint/run/core/healthcheck/healthcheck.go
+++ b/mint/run/core/healthcheck/healthcheck.go
@@ -28,6 +28,7 @@ import (
 	"os"
 	"time"
 
+	jwtgo "github.com/dgrijalva/jwt-go"
 	log "github.com/sirupsen/logrus"
 )
 
@@ -99,7 +100,7 @@ func testLivenessEndpoint(endpoint string) {
 	}
 
 	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: u.Scheme == "https"},
 	}
 	client := &http.Client{Transport: tr, Timeout: timeout}
 	resp, err := client.Get(u.String())
@@ -109,7 +110,7 @@ func testLivenessEndpoint(endpoint string) {
 	}
 	if resp.StatusCode != http.StatusOK {
 		// Status not 200 OK
-		failureLog(function, nil, startTime, "", "GET /minio/health/live returned non OK status", err).Fatal()
+		failureLog(function, nil, startTime, "", fmt.Sprintf("GET /minio/health/live returned %s", resp.Status), err).Fatal()
 	}
 
 	defer resp.Body.Close()
@@ -127,7 +128,7 @@ func testReadinessEndpoint(endpoint string) {
 	}
 
 	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: u.Scheme == "https"},
 	}
 	client := &http.Client{Transport: tr, Timeout: timeout}
 	resp, err := client.Get(u.String())
@@ -138,6 +139,57 @@ func testReadinessEndpoint(endpoint string) {
 	if resp.StatusCode != http.StatusOK {
 		// Status not 200 OK
 		failureLog(function, nil, startTime, "", "GET /minio/health/ready returned non OK status", err).Fatal()
+	}
+
+	defer resp.Body.Close()
+	defer successLogger(function, nil, startTime).Info()
+}
+
+const (
+	defaultPrometheusJWTExpiry = 100 * 365 * 24 * time.Hour
+)
+
+func testPrometheusEndpoint(endpoint string) {
+	startTime := time.Now()
+	function := "testPrometheusEndpoint"
+
+	u, err := url.Parse(fmt.Sprintf("%s%s", endpoint, prometheusPath))
+	if err != nil {
+		// Could not parse URL successfully
+		failureLog(function, nil, startTime, "", "URL Parsing for Healthcheck Prometheus handler failed", err).Fatal()
+	}
+
+	jwt := jwtgo.NewWithClaims(jwtgo.SigningMethodHS512, jwtgo.StandardClaims{
+		ExpiresAt: time.Now().UTC().Add(defaultPrometheusJWTExpiry).Unix(),
+		Subject:   os.Getenv("ACCESS_KEY"),
+		Issuer:    "prometheus",
+	})
+
+	token, err := jwt.SignedString([]byte(os.Getenv("SECRET_KEY")))
+	if err != nil {
+		failureLog(function, nil, startTime, "", "jwt generation failed", err).Fatal()
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: u.Scheme == "https"},
+	}
+	client := &http.Client{Transport: tr, Timeout: timeout}
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		failureLog(function, nil, startTime, "", "Initializing GET request to Prometheus endpoint failed", err).Fatal()
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		// GET request errored
+		failureLog(function, nil, startTime, "", "GET request to Prometheus endpoint failed", err).Fatal()
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		// Status not 200 OK
+		failureLog(function, nil, startTime, "", "GET /minio/prometheus/metrics returned non OK status", err).Fatal()
 	}
 
 	defer resp.Body.Close()
@@ -163,4 +215,5 @@ func main() {
 	// execute tests
 	testLivenessEndpoint(endpoint)
 	testReadinessEndpoint(endpoint)
+	testPrometheusEndpoint(endpoint)
 }

